api/views.py

```diff
@@ -49,6 +51,40 @@ class TimerViewSet(viewsets.ModelViewSet):
     serializer_class = serializers.TimerSerializer
     filterset_class = filters.TimerFilter
 
+    def __timer_operation(self, pk, func):
+        try:
+            timer = models.Timer.objects.get(pk=pk)
+            return func(timer)
+        except models.Timer.DoesNotExist:
+            return Response(
+                {"detail": "timer does not exist"},
+                status=status.HTTP_404_NOT_FOUND
+            )
+
+    @action(detail=True, methods=['patch'])
+    def stop(self, request, pk=None):
+        def do_stop(timer):
+            if not timer.active:
+                return Response(
+                    {"detail": "timer already stopped"},
+                    status=status.HTTP_412_PRECONDITION_FAILED
+                )
+            timer.stop()
+            return Response({"detail": "timer stopped"})
```

Comment:
In keeping with other `PATCH` operations -- lets return a representation of the updated object here instead.